lxd/resources/usb.go

```diff
@@ -0,0 +1,213 @@
+package resources
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/usbid"
+)
+
+var sysBusUSB = "/sys/bus/usb/devices"
+
+// GetUSB returns a filled api.ResourcesUSB struct ready for use by LXD
+func GetUSB() (*api.ResourcesUSB, error) {
+	usb := api.ResourcesUSB{}
+	devices := []api.ResourcesUSBDevice{}
+
+	// List all USB devices
+	entries, err := ioutil.ReadDir(sysBusUSB)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to list %q", sysBusUSB)
+	}
```

Comment:
This will break on a system without a USB bus...

Reply:
Fixed